### PR TITLE
Add a manifest key for bundle metadata

### PR DESCRIPTION
**Pull request by viovanov**
This tries to implement #3431.

### CODE REVIEW

File: bundle/store.go [hunk 1]
```diff
@@ -222,6 +226,27 @@ func readRevisionFromStore(ctx context.Context, store storage.Store, txn storage
 	return str, nil
 }
 
+// ReadBundleRevisionFromStore returns the metadata in the specified bundle.
```
Comment (anderseknert) [inline bundle/store.go:229]: Wrong name here

File: bundle/store.go [hunk 2]
```diff
@@ -222,6 +226,27 @@ func readRevisionFromStore(ctx context.Context, store storage.Store, txn storage
...
+// If the bundle is not activated, this function will return
+// storage NotFound error.
+func ReadBundleMetadataFromStore(ctx context.Context, store storage.Store, txn storage.Transaction, name string) (map[string]interface{}, error) {
+	return readMetadataFromStore(ctx, store, txn, metadataPath(name))
```
Comment (anderseknert) [inline bundle/store.go:233]: Why not just inline this?

File: bundle/store.go [hunk 3]
```diff
@@ -222,6 +226,27 @@ func readRevisionFromStore(ctx context.Context, store storage.Store, txn storage
 	return str, nil
 }
 
+// ReadBundleRevisionFromStore returns the metadata in the specified bundle.
+// If the bundle is not activated, this function will return
+// storage NotFound error.
+func ReadBundleMetadataFromStore(ctx context.Context, store storage.Store, txn storage.Transaction, name string) (map[string]interface{}, error) {
```
Comment (anderseknert) [inline bundle/store.go:232]: I don't see any test calling this. Is it called indirectly?

File: bundle/store.go (hunk 2 above)
Comment (viovanov) [reply bundle/store.go:233]: Following the same pattern as `ReadBundleRevisionFromStore`.

File: bundle/store.go (hunk 3 above)
Comment (viovanov) [reply bundle/store.go:232]: No, thanks for the hint, I've updated tests.

**Review by ashutosh-narkar — COMMENTED** [review]
@viovanov thanks for the contribution ! We probably need to test this but wouldn't it be possible to include metadata that violates the bundle's roots ? Also maybe I'm missing something but why can't we include this in the data under a key `metadata` ?

**Comment by tsandall** [comment]
Just reviewed the changes. LGTM modulo a documentation update. @viovanov, if you can update [this](https://github.com/open-policy-agent/opa/blob/main/docs/content/management-bundles.md#bundle-file-format) section in the docs, that would be perfect.

@ashutosh-narkar I'm not sure I follow your concern about violating the bundle roots. Can you explain? Also, what do you mean by `include this in the data under a key metadata`?

**Comment by viovanov** [comment]
@tsandall I've updated the doc - hope that's ok, couldn't think of anything else to write.

**Comment by tsandall** [comment]
@viovanov LGTM!

**Comment by tsandall** [comment]
@viovanov can you squash your commits? We try to avoid including merge commits in the PR. There should just be one but I see three.

**Comment by viovanov** [comment]
@tsandall done, thanks!

**Review by tsandall — APPROVED** [review]

**Review by ashutosh-narkar — APPROVED** [review]

**Comment by viovanov** [comment]
@tsandall @ashutosh-narkar did I introduce a race?

**Comment by tsandall** [comment]
no, this looks like a flake due to some of my recent changes.

**Pull request merged**